lib/rbac/filterer.rb

```diff
@@ -438,11 +434,12 @@ def to_class(klass)
       klass.kind_of?(String) || klass.kind_of?(Symbol) ? klass.to_s.constantize : klass
     end
 
-    def method_with_scope(ar_scope, options)
-      ar_scope.apply_legacy_finder_options(options)
-    end
-
     def apply_scope(klass, scope)
+      klass = to_class(klass)
+
+      # could just call all on everything, but that will display deprecation warnings
+      klass = klass.all if klass < ActiveRecord::Base || klass.kind_of?(ActsAsArModel)
```

Comment:
I have tested this locally and it fails:
  it "#check_acts_as_ar_model" do
    klass = LiveMetric
    expect(klass.kind_of?(ActsAsArModel)).to be(true)
  end
It seems that klass is of type Class and kind_of is checking some of ancestor of that class instead LiveMetric.
[Class, Module, RequireNested, MoreCoreExtensions::ModuleNamespace, Vmdb::ClassLogging, ActiveSupport::Dependencies::ModuleConstMissing, Module::Concerning, NakayoshiFork, ActiveSupport::ToJsonWithActiveSupportEncoder, Object, AvailabilityMixin::ClassMethods, AvailabilityMixin, PP::ObjectMixin, Vmdb::GlobalMethods, MoreCoreExtensions::ObjectNamespace, GettextI18nRails::HtmlSafeTranslations, FastGettext::Translation, ActiveSupport::Dependencies::Loadable, JSON::Ext::Generator::GeneratorMethods::Object, ActiveSupport::Tryable, Kernel, BasicObject]